$NetBSD: patch-af,v 1.2 2012/05/10 22:11:20 dholland Exp $

- don't declare own errno
- fix build failure where gzgetc() is a macro (seen in a recent Linux
  bulk build)

XXX: what this code is doing (mixing gzFile* and FILE* indiscriminately
XXX: without even casting) is vile and will probably stop compiling at
XXX: some point.

--- misc/archive.c.orig	2004-12-12 01:16:54.000000000 +0000
+++ misc/archive.c
@@ -48,9 +48,6 @@ Foundation, Inc., 675 Mass Ave, Cambridg
 #define MAXBUFSIZE 32768
 
 
-extern int errno;
-
-
 int
 fsizeof (const char *filename)
 // If USE_ZLIB is defined this function is very slow. Please avoid to use
@@ -498,7 +495,7 @@ fgetc2 (FILE *file)
   if (fmode == FM_NORMAL)
     return fgetc (file);
   else if (fmode == FM_GZIP)
-    return gzgetc (file);
+    return gzgetc ((struct gzFile *)file);
   else if (fmode == FM_ZIP)
     {
       char c;
